mongo/options/clientoptions.go

```diff
@@ -300,187 +322,305 @@ func (c *ClientOptions) ApplyURI(uri string) *ClientOptions {
 }
 
-// SetPoolMonitor specifies the PoolMonitor for a server's connection pool.
+// SetPoolMonitor specifies a PoolMonitor to receive connection pool events. See the event.PoolMonitor documentation
+// for more information about the structure of the monitor and events that can be received.
 func (c *ClientOptions) SetPoolMonitor(m *event.PoolMonitor) *ClientOptions {
 	c.PoolMonitor = m
 	return c
 }
 
-// SetMonitor specifies a command monitor used to see commands for a client.
+// SetMonitor specifies a CommandMonitor to receive command events. See the event.CommandMonitor documentation for more
+// information about the structure of the monitor and events that can be received.
 func (c *ClientOptions) SetMonitor(m *event.CommandMonitor) *ClientOptions {
 	c.Monitor = m
 	return c
 }
 
-// SetReadConcern specifies the read concern.
+// SetReadConcern specifies the read concern to use to configure the Client. A read concern level can also be set
+// through the "readConcernLevel" URI option (e.g. "readConcernLevel=majority"). The default is nil, meaning the server
+// will use its configured default.
 func (c *ClientOptions) SetReadConcern(rc *readconcern.ReadConcern) *ClientOptions {
 	c.ReadConcern = rc
 
 	return c
 }
 
-// SetReadPreference specifies the read preference.
+// SetReadPreference specifies the read preference to use to configure the Client. This can also be set through the
+// following URI options:
+//
+// 1. "readPreference" - Specifiy the read preference mode (e.g. "readPreference=primary").
+//
+// 2. "readPreferenceTags": Specify one or more read preference tags
+// (e.g. "readPreferenceTags=region:south,datacenter:A").
+//
+// 3. "maxStalenessSeconds" (or "maxStaleness"): Specify a maximum replication lag for reads from secondaries in a
+// replica set (e.g. "maxStalenessSeconds=10").
+//
+// The default is readpref.Primary(). See https://docs.mongodb.com/manual/core/read-preference/#read-preference for
+// more information about read preferences.
 func (c *ClientOptions) SetReadPreference(rp *readpref.ReadPref) *ClientOptions {
 	c.ReadPreference = rp
 
 	return c
 }
 
-// SetRegistry specifies the bsoncodec.Registry.
+// SetRegistry specifies the BSON registry to use to configure the Client. The default is bson.DefaultRegistry.
 func (c *ClientOptions) SetRegistry(registry *bsoncodec.Registry) *ClientOptions {
 	c.Registry = registry
 	return c
 }
 
-// SetReplicaSet specifies the name of the replica set of the cluster.
+// SetReplicaSet specifies the replica set name for the cluster. If specified, the cluster will be treated as a replica
+// set and the driver will automatically discover all nodes in the set, starting with the nodes specified through
+// ApplyURI or SetHosts. All nodes in the replica set must have the same replica set name, or they will not be
+// considered as part of the set by the Client. This can also be set through the "replicaSet" URI option (e.g.
+// "replicaSet=replset"). The default is empty.
```

Comment:
Specify interaction between this option and connect=direct.

Reply:
Right now, I think the behavior is undefined, but it should be a client-side error. I filed SPEC-1438 a while ago to deal with some of this (see the first comment on the ticket) and that was closed as a dup of SPEC-1248, which is scheduled to be done. I can make sure to talk to the spec author to make sure we specify the interaction between number of hosts, value of `replicaSet`, and value of `connect`.